crates/database/src/onchain_invalidations.rs

```diff
@@ -0,0 +1,169 @@
+use crate::{events::EventIndex, OrderUid, PgTransaction};
+use sqlx::{Executor, PgConnection};
+
+#[derive(Clone, Debug, Default, sqlx::FromRow, Eq, PartialEq)]
+pub struct OnchainInvalidationRow {
+    pub uid: OrderUid,
+    pub block_number: i64,
+    pub log_index: i64,
+}
+
+pub async fn append_invalidations(
+    ex: &mut PgTransaction<'_>,
+    events: &[(EventIndex, OrderUid)],
+) -> Result<(), sqlx::Error> {
+    for (index, event) in events {
+        insert_onchain_invalidation(ex, index, event).await?;
+    }
+    Ok(())
+}
+
+pub async fn delete_invaldiations(
+    ex: &mut PgTransaction<'_>,
+    block_number: i64,
+) -> Result<(), sqlx::Error> {
+    const QUERY_INVALIDATION: &str = "DELETE FROM onchain_order_invalidations \
+                                      WHERE block_number >= $1;";
+    ex.execute(sqlx::query(QUERY_INVALIDATION).bind(block_number))
+        .await?;
+    Ok(())
+}
+
+pub async fn insert_onchain_invalidation(
```

Comment:
super nit: It's a bit strange to expose 2 functions in the module which do basically the same thing but have different names:
`insert_onchain_invalidation()` and `append_invalidations()`.
Do you think it would be nice to only expose 1 function or maybe to give them similar names?